src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionOptions.cs

```diff
@@ -3,113 +3,31 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
-using System.Globalization;
 using System.IO;
-using System.Text;
 
 namespace Microsoft.Data.Common
 {
     internal partial class DbConnectionOptions
     {
-        // instances of this class are intended to be immutable, i.e readonly
-        // used by pooling classes so it is much easier to verify correctness
-        // when not worried about the class being modified during execution
-
-        public DbConnectionOptions(string connectionString, Dictionary<string, string> synonyms)
-        {
-            _parsetable = new Dictionary<string, string>();
-            _usersConnectionString = ((null != connectionString) ? connectionString : "");
-
-            // first pass on parsing, initial syntax check
-            if (0 < _usersConnectionString.Length)
-            {
-                _keyChain = ParseInternal(_parsetable, _usersConnectionString, true, synonyms, false);
-                HasPasswordKeyword = (_parsetable.ContainsKey(KEY.Password) || _parsetable.ContainsKey(SYNONYM.Pwd));
-                HasUserIdKeyword = (_parsetable.ContainsKey(KEY.User_ID) || _parsetable.ContainsKey(SYNONYM.UID));
-            }
-        }
-
-        protected DbConnectionOptions(DbConnectionOptions connectionOptions)
-        { // Clone used by SqlConnectionString
-            _usersConnectionString = connectionOptions._usersConnectionString;            
-            _parsetable = connectionOptions._parsetable;
-            _keyChain = connectionOptions._keyChain;
-            HasPasswordKeyword = connectionOptions.HasPasswordKeyword;
-            HasUserIdKeyword = connectionOptions.HasUserIdKeyword;
-        }
-
-        public bool IsEmpty => _keyChain == null;
-
-        internal bool TryGetParsetableValue(string key, out string value) => _parsetable.TryGetValue(key, out value);
-
-        // same as Boolean, but with SSPI thrown in as valid yes
-        public bool ConvertValueToIntegratedSecurity()
+        internal string ExpandAttachDbFileName(string replacementValue)
         {
-            string value;
-            return _parsetable.TryGetValue(KEY.Integrated_Security, out value) && value != null ?
-                ConvertValueToIntegratedSecurityInternal(value) :
-                false;
-        }
+            int copyPosition = 0;
 
-        internal bool ConvertValueToIntegratedSecurityInternal(string stringValue)
-        {
-            if (CompareInsensitiveInvariant(stringValue, "sspi") || CompareInsensitiveInvariant(stringValue, "true") || CompareInsensitiveInvariant(stringValue, "yes"))
-                return true;
-            else if (CompareInsensitiveInvariant(stringValue, "false") || CompareInsensitiveInvariant(stringValue, "no"))
-                return false;
-            else
+            System.Text.StringBuilder builder = new(_usersConnectionString.Length);
```

Comment:
NIT: could this be simplified by adding the namespace at the top? Code will look cleaner in general and objects will be categorized in a better way.